tools/c7n_azure/c7n_azure/dependency_manager.py

```diff
@@ -0,0 +1,146 @@
+import hashlib
+import json
+import logging
+import os
+import re
+import subprocess
+import sys
+from builtins import bytes
+
+logger = logging.getLogger('c7n_azure.dependency_manager')
+
+
+class DependencyManager(object):
+
+    @staticmethod
+    def _run(cmd, verbose=False, **kwargs):
+        if verbose:
+            stdout = stderr = None
+        else:
+            stdout = stderr = subprocess.PIPE
+
+        logger.debug(' '.join(cmd))
+        return subprocess.run(cmd, stdout=stdout, stderr=stderr, **kwargs)
+
+    @staticmethod
+    def get_dependency_packages_list(packages, excluded_packages):
+        try:
+            from pip._internal.utils.misc import get_installed_distributions
+        except ImportError:
+            from pip import get_installed_distributions
+
+        dists = get_installed_distributions()
+        res = []
+        for p in packages:
+            res.extend([str(r) for r in next(d.requires() for d in dists if (p + ' ') in str(d))])
+
+        res = [t for t in res if not any((e in t) for e in excluded_packages + packages)]
+
+        # boto3 is a dependency for both c7n and c7n_mailer.. Remove the duplicate from the list
+        # because not all versions of pip can handle this.
+        res = [t for t in res if not any((e != t and t in e) for e in res)]

```

Comment:
boto3 & botocore shouldn't be a runtime requirement for c7n_azure functions, if it i think its a bug and happy to fix. it hasn't been a req for gcp cloud funcs, please file an issue if there are any details.

Reply:
I was trying to run custodian without boto3 & botocore installed ~1 month ago and back then it didn't work.. I'll definitely take another look